src/metabase/models/secret.clj

```diff
@@ -62,9 +62,11 @@
 
   `driver?` is an optional argument that is only used if an ostensibly existing file value (i.e. `:file-path`) can't be
   resolved, in order to render a more user-friendly error message (by looking up the display names of the connection
-  properties involved)."
+  properties involved).
+
+  `ext?` is an optional argument that sets the file extension used for the temporary file, if one needs to be created."
   {:added "0.42.0"}
-  ^File [{:keys [connection-property-name id value] :as secret} driver?]
+  ^File [{:keys [connection-property-name id value] :as secret} & [driver? ext?]]
```

Comment:
Varag signatures make linters less effective.

Reply:
Fair, I'll make it multi-arity instead